docs/enhancements/image-handling-service.md

```diff
@@ -0,0 +1,165 @@
+assisted service endpoint.
+
+The assisted service will retain the `POST /clusters/{cluster_id}/downloads/image`
+endpoint which will save the image information to its database and make it available
+to the image service. In the case of the kube API, the controllers will continue
+to save image information from the various CRs as they do currently.
+
+### User Stories
+
+This change should be transparent to users. As such, stories will focus on
+deployment and management concerns rather than user-facing changes.
+
+#### Story 1
+
+An additional deployment and service will be managed for all deployment methods.
+This includes SaaS, operator, and CI (minikube). Whether an additional image
+is still an open question.
+
+### Implementation Details/Notes/Constraints
+
+#### Authentication and Authorization
+
+The image service will not need to implement authentication or authorization
+directly as it doesn't manage any user data. The user will provide a token to
+the image service and the image service will directly pass that token to
+assisted service on each call. Assisted service will validate the token before
+giving the image service any user information.
+
+#### Template Images
+
+Currently, when the assisted service starts, it downloads and caches each RHCOS
+image specified in the `OPENSHIFT_VERSIONS` environment variable. It also
+creates a minimal iso image for each version by removing the root filesystem.
+These "template" images are the base for creating the discovery iso.
+
+The image service will take on the responsibility of managing the RHCOS template
+images and creating the minimal ISO template images. This means that the image
+service will require persistent storage (when not using S3).
+
+#### Image Streaming
+
+The version of the image service described here would require editing the image
+while streaming the download rather than the current behavior which stores the
+customized image to be downloaded in a separate call. Allowing the image to be
+stored would complicate the proposal and is discussed in the "Alternatives"
+section.
+
+#### Assisted Service API Changes
+
+Some additional details need to be exposed over the assisted service API to
+expose all the information the image service needs to create the discovery ISO.
+Specifically, the values that are required and not currently accessible over
+the API are:
+- Pull secret
+
+### Risks and Mitigations
+
+Relying strictly on streaming will increase the load on the image service in
+the SaaS deployment as presigned links directly to S3 would no longer be provided.
+Downloading the image through the service is already required when running
+on a local cluster so this shouldn't be a significant problem, but should
+be load tested. This will be further mitigated by splitting the service as
+many downloads won't affect the other operations of assisted service.
+
+
+## Design Details [optional]
+
+
+### Open Questions
```

Comment:
What about supporting ranges? (download specific section) i remember that there was a discussion about it but i don't remember what was the conclusion.

Reply:
I mentioned that in [Drawbacks](https://github.com/openshift/assisted-service/pull/2172/files#diff-35e94f065c5c812e0d0a2a0c7fb913b43d795f0d4b801f04aa0a643e843e6664R152-R153) and [Non-Goals](https://github.com/openshift/assisted-service/pull/2172/files#diff-35e94f065c5c812e0d0a2a0c7fb913b43d795f0d4b801f04aa0a643e843e6664R36).
I didn't call out why serving to a BMC would be more difficult, but this is the reason.

Generally we'd have to have the whole file accessible to support range queries. It's probably possible to do without, but it would be much more complicated.

Reply:
From a recent conversation with @zaneb it sounds like supporting range access (even if just in the library) would be useful for them. This is a potential point for collaboration.